core/codec/cbor/cbor_dump.cpp

```diff
@@ -15,28 +15,29 @@ namespace fc {
   }
 
   std::string dumpCid(const CID &cid) {
-    auto &mh{cid.content_address};
+    const auto &mh{cid.content_address};
     auto hash{mh.getHash()};
     if (cid.version == CID::Version::V1
         && cid.content_type == CID::Multicodec::DAG_CBOR
         && mh.getType() == libp2p::multi::HashType::blake2b_256) {
       return dumpBytes(hash);
     }
     OUTCOME_EXCEPT(bytes, cid.toBytes());
-    return dumpBytes(
-               gsl::make_span(bytes).subspan(0, bytes.size() - hash.size() - 1))
+    return dumpBytes(gsl::make_span(bytes).subspan(
+               0, static_cast<ptrdiff_t>(bytes.size() - hash.size() - 1)))
            + "_" + dumpBytes(hash);
   }
 
   struct LessCborKey {
     bool operator()(const std::string &l, const std::string &r) const {
```

Comment:
```suggestion
    return less(lhs.size(), rhs.size(), lhs, rhs);
```
(looks similar to `LessCborKey` from cbor_encode_stream.cpp, except span cast, it can have two `operator()()` for bytes and string_view, if needed)